SSLException: Received fatal alert: protocol_version for running testsuite in FIPS mode

_master_ (commit 98fe84975b90e21593ff16be687c1a9ad7ff2f20)

Failed on FIPS build: http://jenkins.hazelcast.com/view/FIPS/job/Hazelcast-3.x-IbmJDK8-fips-nightly/3/testReport/

Following tests failed:
* `com.hazelcast.client.starter.HazelcastClientStarterTest.testClientMap_async`
* `com.hazelcast.client.starter.HazelcastClientStarterTest.testClientLifecycle`
* `com.hazelcast.client.starter.HazelcastClientStarterTest.testAdvancedClientMap`
* `com.hazelcast.client.starter.HazelcastClientStarterTest.testClientMap`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testMultiMapService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testMapService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testClusterService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testHazelcastInstanceImpl`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testSetService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testNode`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testNodeEngine`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testCacheService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testQueueService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testSerializationService`
* `com.hazelcast.test.starter.answer.test.AnswerTest.testPartitionService`
* `com.hazelcast.test.starter.constructor.test.MergePolicyProviderConstructorTest.testConstructor`
* `com.hazelcast.test.starter.test.HazelcastProxyFactoryTest.testProxyHazelcastInstanceClasses_ofSameVersion_areSame`
* `com.hazelcast.test.starter.test.HazelcastStarterTest.testMemberWithConfig_withFirewallingConnectionManager`
* `com.hazelcast.test.starter.test.HazelcastStarterTest.testMember`
* `com.hazelcast.test.starter.test.HazelcastStarterTest.testHazelcastInstanceCompatibility_withStarterInstance`
* `com.hazelcast.test.starter.test.HazelcastStarterTest.testMemberWithConfig_withoutFirewallingConnectionManager`
* `com.hazelcast.test.starter.test.HazelcastVersionLocatorTest.testDownloadVersion`
* `com.hazelcast.test.starter.test.PatchLevelCompatibilityTest.testAll_V38_Versions`
* `com.hazelcast.test.starter.test.PatchLevelCompatibilityTest.testAll_V39_Versions`
* `com.hazelcast.test.starter.test.PatchLevelCompatibilityTest.testAll_V37_Versions`
* `com.hazelcast.test.starter.test.PatchLevelCompatibilityTest.testMap_whenMixed_V37_Cluster`
* `com.hazelcast.test.starter.test.PatchLevelCompatibilityTest.testAll_V310_Versions`

Stacktrace is like:
```
com.hazelcast.test.starter.GuardianException: javax.net.ssl.SSLException: Received fatal alert: protocol_version
	at com.hazelcast.test.starter.HazelcastStarterUtils.rethrowGuardianException(HazelcastStarterUtils.java:53)
	at com.hazelcast.test.starter.HazelcastVersionLocator.downloadFile(HazelcastVersionLocator.java:151)
	at com.hazelcast.test.starter.HazelcastVersionLocator.downloadMember(HazelcastVersionLocator.java:114)
	at com.hazelcast.test.starter.HazelcastVersionLocator.locateMember(HazelcastVersionLocator.java:79)
	at com.hazelcast.test.starter.HazelcastVersionLocator.locateVersion(HazelcastVersionLocator.java:62)
	at com.hazelcast.test.starter.HazelcastStarter$HazelcastVersionClassloaderFuture.get(HazelcastStarter.java:343)
	at com.hazelcast.test.starter.HazelcastStarter.getTargetVersionClassloader(HazelcastStarter.java:210)
	at com.hazelcast.test.starter.HazelcastStarter.newHazelcastInstance(HazelcastStarter.java:101)
	at com.hazelcast.test.starter.HazelcastStarter.newHazelcastInstance(HazelcastStarter.java:69)
	at com.hazelcast.client.starter.HazelcastClientStarterTest.testClientMap_async(HazelcastClientStarterTest.java:94)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:811)
Caused by: javax.net.ssl.SSLException: Received fatal alert: protocol_version
	at com.ibm.jsse2.k.a(k.java:5)
	at com.ibm.jsse2.k.a(k.java:23)
	at com.ibm.jsse2.av.b(av.java:343)
	at com.ibm.jsse2.av.a(av.java:981)
	at com.ibm.jsse2.av.i(av.java:869)
	at com.ibm.jsse2.av.a(av.java:19)
	at com.ibm.jsse2.av.startHandshake(av.java:672)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:261)
	at org.apache.http.impl.conn.HttpClientConnectionOperator.connect(HttpClientConnectionOperator.java:118)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:314)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:357)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:218)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:194)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:85)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:108)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:106)
	at com.hazelcast.test.starter.HazelcastVersionLocator.downloadFile(HazelcastVersionLocator.java:140)
	... 20 more
```

--- a/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java
@@ -18,22 +18,20 @@ package com.hazelcast.test.starter;
 
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.logging.Logger;
-import org.apache.http.HttpEntity;
-import org.apache.http.client.methods.CloseableHttpResponse;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
 
+import java.io.BufferedInputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
 
 import static com.hazelcast.nio.IOUtil.closeResource;
+import static com.hazelcast.nio.IOUtil.drainTo;
 import static com.hazelcast.test.JenkinsDetector.isOnJenkins;
 import static com.hazelcast.test.starter.HazelcastStarterUtils.rethrowGuardianException;
 import static java.io.File.separator;
 import static java.lang.String.format;
-import static org.apache.http.HttpStatus.SC_OK;
 
 public class HazelcastVersionLocator {
 
@@ -127,32 +125,23 @@ public class HazelcastVersionLocator {
     }
 
     private static File downloadFile(String url, File targetDirectory, String filename) {
-        CloseableHttpClient client = HttpClients.createDefault();
         File targetFile = new File(targetDirectory, filename);
         if (targetFile.isFile() && targetFile.exists()) {
             return targetFile;
         }
-        HttpGet request = new HttpGet(url);
-
-        CloseableHttpResponse response = null;
         FileOutputStream fos = null;
+        InputStream is = null;
         try {
-            response = client.execute(request);
-            if (response.getStatusLine().getStatusCode() != SC_OK) {
-                throw new GuardianException("Cannot download file from " + url
-                        + ", http response code: " + response.getStatusLine().getStatusCode());
-            }
-            HttpEntity entity = response.getEntity();
+            is = new BufferedInputStream(new URL(url).openStream());
             fos = new FileOutputStream(targetFile);
-            entity.writeTo(fos);
+            drainTo(is, fos);
             targetFile.deleteOnExit();
             return targetFile;
         } catch (IOException e) {
             throw rethrowGuardianException(e);
         } finally {
             closeResource(fos);
-            closeResource(response);
-            closeResource(client);
+            closeResource(is);
         }
     }
 

